lib/ClangImporter/ImportMacro.cpp

```diff
@@ -334,6 +345,29 @@ static ValueDecl *importMacro(ClangImporter::Implementation &impl,
         tokenI[3].is(clang::tok::r_paren)) {
       return importStringLiteral(impl, DC, macro, name, tokenI[2],
                                  MappedStringLiteralKind::CFString, ClangN);
+    // Check for literals with a cast, e.g. "(cpu_type_t) 1"
+    } else if (tokenI[0].is(clang::tok::l_paren) &&
+               tokenI[1].is(clang::tok::identifier) &&
```

Comment:
Shouldn't the cast be parsed before we go into this switch, so the cast type can be setup early and apply to any of the cases where it makes sense? It would be more general, then.